docs/usability/creating_invocable.md

```diff
@@ -0,0 +1,685 @@
+
+One final import is needed that will seem a little strange.
+```python
+import DriverBuilder
+```
+Stuart works in such a way that it expects your invokable to be running in the namespace that is named in.
+If you run your builder directly from the commandline, it will be running in \_\_main__, which can cause problems.
+**In a nutshell, you'll need to import the name of your file.**
+We'll see where this is used at the end.
+
+Now the settings class!
+
+```python
+class BinaryBuildSettingsManager():
+    ''' Platform settings will be accessed through this implementation. '''
+
+    def GetActiveScopes(self):
+        ''' get scope '''
+        raise NotImplementedError()
+
+    def GetWorkspaceRoot(self):
+        ''' get WorkspacePath '''
+        raise NotImplementedError()
+
+    def GetPackagesPath(self):
+        pass
+
+     def GetName(self):
+        ''' Get the name of the repo, platform, or product being build by CI '''
+        raise NotImplementedError()
+
+    def AddCommandLineOptions(self, parserObj):
+        ''' Implement in subclass to add command line options to the argparser '''
+        pass
+
+    def RetrieveCommandLineOptions(self, args):
+        '''  Implement in subclass to retrieve command line options from the argparser '''
+        pass
+
+    def GetModulePkgsPath(self):
+        ''' Get the modules that we care about '''
+        return ""
+
+```
+
+We've implemented a few methods that are needed to get the SDE off the ground.
+- _GetActiveScopes_ is needed to init the SDE. This cause different plugins to load or not.
+- _GetWorkspaceRoot_ gets the folder that is the root of your workspace.
+- _GetPackagePaths_ gets the folder locations that you will resolve EDK2 paths against.
+- _GetName_ is the thing we are building
+- _AddCommandLineOptions_ gives our settings the chance to set items in the parser object
+- _RetrieveCommandLineOptions_ gives us the chance to read the arguments from the commandline
+
+Now that we have our base methods, let's add one more to control the configurations we are going to build.
+
+```python
+
+class BinaryBuildSettingsManager():
+    ''' Platform settings will be accessed through this implementation. '''
+
+    ....
+
+    def GetConfigurations(self):
+        '''
+        Gets the next configuration of this run
+        This is a generator pattern - use yield
+        '''
+        raise NotImplementedError()
+```
+
+- _GetConfigurations_ is our way to get the configurations we want to build. We'll use a generator/iterator pattern here that we'll see later.
+
+We also need some methods to have callbacks into various stages of the process so that we can do nuget commands and prepare the nuget package.
+
+```python
+
+class BinaryBuildSettingsManager():
+    ''' Platform settings will be accessed through this implementation. '''
+
+    ....
+
+     def PreFirstBuildHook(self):
+        ''' Called after the before the first build '''
```

Comment:
what?  I think my issue is with the comment not the function

Reply:
Fixed